src/mod_disco.erl

```diff
@@ -94,118 +92,193 @@ config_spec() ->
 
 server_info_spec() ->
     #section{
-       items = #{<<"name">> => #option{type = string,
+       items = #{<<"name">> => #option{type = binary,
                                        validate = non_empty},
-                 <<"urls">> => #list{items = #option{type = string,
+                 <<"urls">> => #list{items = #option{type = binary,
                                                      validate = url}},
                  <<"modules">> => #list{items = #option{type = atom,
                                                         validate = module}}
                 },
-       required = [<<"name">>, <<"urls">>],
-       process = fun ?MODULE:process_server_info/1
+       required = [<<"name">>, <<"urls">>]
       }.
 
-process_server_info(KVs) ->
-    {[[{name, Name}], [{urls, URLs}]], _} = proplists:split(KVs, [name, urls]),
-    Modules = proplists:get_value(modules, KVs, all),
-    {Modules, Name, URLs}.
+supported_features() -> [dynamic_domains].
 
 %% IQ handlers
 
--spec process_local_iq_items(jid:jid(), jid:jid(), mongoose_acc:t(), jlib:iq()) ->
+-spec process_local_iq_items(mongoose_acc:t(), jid:jid(), jid:jid(), jlib:iq(), map()) ->
     {mongoose_acc:t(), jlib:iq()}.
-process_local_iq_items(_From, _To, Acc, #iq{type = set, sub_el = SubEl} = IQ) ->
+process_local_iq_items(Acc, _From, _To, #iq{type = set, sub_el = SubEl} = IQ, _Extra) ->
     {Acc, IQ#iq{type = error, sub_el = [SubEl, mongoose_xmpp_errors:not_allowed()]}};
-process_local_iq_items(From, To, Acc, #iq{type = get, lang = Lang, sub_el = SubEl} = IQ) ->
+process_local_iq_items(Acc, From, To, #iq{type = get, lang = Lang, sub_el = SubEl} = IQ, _Extra) ->
+    HostType = mongoose_acc:host_type(Acc),
     Node = xml:get_tag_attr_s(<<"node">>, SubEl),
-    Host = To#jid.lserver,
-    case mongoose_hooks:disco_local_items(Host, From, To, Node, Lang) of
-        {result, Items} ->
+    case mongoose_disco:get_local_items(HostType, From, To, Node, Lang) of
+        empty ->
+            Error = mongoose_xmpp_errors:item_not_found(),
```

Comment:
maybe adding a text like `mongoose_xmpp_errors:item_not_found(<<"en">>, <<"Local disco items not found">>)`.

Reply:
I am not sure if we need `en` as we have `Lang` here... anyway, there are more such errors in `mod_disco`, let's address it in a separate PR.